ocs_ci/ocs/ui/base_ui.py

```diff
@@ -1485,20 +1500,1136 @@ def __init__(self):
+            Each alert level is associated with a list of alert titles.
+
+        Note:
+            Alerts tab should be opened.
+            The alert levels are expanded to read the titles and then shrunk back.
+
+        """
+        logger.info("reading alerts from the alert tab")
+        alerts_dict = dict()
+        alerts_lvl_to_num = self.get_number_of_alerts()
+        alert_levels_exists = [
+            alert for alert, value in alerts_lvl_to_num.items() if value > 0
+        ]
+        for alert_lvl in alert_levels_exists:
+
+            # expand each alert and read number of alerts
+            # to work with large number of alerts need to scroll and pick all titles
+            from ocs_ci.ocs.ui.helpers_ui import format_locator
+
+            self.do_click(
+                format_locator(self.topology_loc["alert_list_expand_arrow"], alert_lvl)
+            )
+            alerts_titles = self.get_elements(
+                self.topology_loc["alerts_sidebar_alert_title"]
+            )
+
+            list_of_alert_per_alert_lvl = []
+            for alert_titles in alerts_titles:
+                alert_text = alert_titles.text
+                # check if alert has text, it may be graphical element such as "!"-icon
+                if alert_text.strip():
+                    list_of_alert_per_alert_lvl.append(alert_text)
+
+            alerts_dict[alert_lvl] = list_of_alert_per_alert_lvl
+            # shrink alert lvl back
+            self.do_click(
+                format_locator(self.topology_loc["alert_list_expand_arrow"], alert_lvl)
+            )
+        logger.info(f"\n{json.dumps(alerts_dict, indent=4)}")
+        return alerts_dict
+
+    def open_details_tab(self):
+        """
+        Opens the details tab in the UI.
+        """
+        self.do_click(self.topology_loc["details_sidebar_tab"], enable_screenshot=True)
+        logger.info("Details tab is open")
+
+    def open_resources_tab(self):
+        """
+        Opens the resources tab in the UI.
+        """
+        self.do_click(
+            self.topology_loc["resources_sidebar_tab"], enable_screenshot=True
+        )
+        logger.info("Resources tab is open")
+
+    def open_observe_tab(self):
+        """
+        Opens the observe tab in the UI.
+        """
+        self.do_click(self.topology_loc["observe_sidebar_tab"], enable_screenshot=True)
+        logger.info("Observe tab is open")
+
+    def get_number_of_alerts(self):
+        """
+        Retrieves the number of alerts categorized by severity level.
+
+        Returns:
+            dict: Dictionary containing the number of alerts for each severity level:
+                  {'Critical': <critical_alerts>, 'Warning': <warning_alerts>, 'Info': <info_alerts>}
+        """
+        alerts_dict = {"Critical": -1, "Warning": -1, "Info": -1}
+        for alert_lvl, val in alerts_dict.items():
+            from ocs_ci.ocs.ui.helpers_ui import format_locator
```

Comment:
Can you take the import out of the loop?

Reply:
yes) good point